File: HW7/driver2.java
import ben.Tetromino;
import ben.Tetris;
import ben.block_type;

import java.util.Scanner;

/**
 * Test driver2
 */
public class driver2{
    public static void main(String[] args) throws InterruptedException {
        char a;
        System.out.printf("------------CONSTRUCTORS---------\n");
        System.out.println("I will create a Tetris board 10X10\n");
        Tetris board = new Tetris(10,10);
        Tetris board_temp = new Tetris(10,10);

        board.get_Tetro_Obj().setter(block_type.Z);
        System.out.println("\nI will set Tetromino with Z");
        board.get_Tetro_Obj().print();

        System.out.println("\n--------Rotate And Print Function---------");
        board.get_Tetro_Obj().rotate('r');
        board.get_Tetro_Obj().print();

        System.out.println("------------Draw Function --------------\n");
        board.draw();

        System.out.println("---------------Size Functions----------------\n");
        System.out.printf("Number of row: %d\n", board.GetLenOfRow());
        System.out.printf("Number of column: %d\n", board.GetLenOfClmn());

        System.out.println("----------Add Function With Is game over function----------------\n");
        System.out.println("I will add Z tetromino to top");
        board.is_game_over(board.get_Tetro_Obj());
        board.draw();

        System.out.println("---------Validity And Add Functions -------------");
        System.out.println("T tetromino will be added at point 7, 5 if this point is valid for Tetromino\n");
        board.get_Tetro_Obj().setter(block_type.T);
        if(board.isvalid_place(board.get_Tetro_Obj(), 7, 5))
            board.add(board.get_Tetro_Obj(), 7, 5);
        board.draw();

        System.out.println("\n\nI will clean the board by using clone function");
        board.clone_board(board_temp);
        board.draw();

        System.out.println("\n---------------Move And Animate Functions-----------");
        System.out.println("T tetromino will go right 3 times and it will be animated\n");
        System.out.println("Enter a to continue and see animate\n");

        Scanner sc = new Scanner(System.in);
        a = sc.next().charAt(0);

        board.move('r', 3, board_temp);


    }
}

File: HW3/driver2.cpp
#include "Tetris.hpp"

int main()
{
    int len;
    cout<< "This is 10X10 board\n";
    Tetris board_obj(10, 10);
    Tetris temp(10, 10);
    for(int i=0; i<10; i++)
    {
        for(int j=0; j<10; j++)
            temp.getBoard()[i][j] = board_obj.getBoard()[i][j];
    }
    cout <<"I create tetromino j\n";
    board_obj.get_Tetro_Obj().setter(block_type::J);
    cout << "I will print tetromino\n";
    board_obj.get_Tetro_Obj().print();
    cout << "And now, i will print tetromino by using getShapeOfTetris func\n";
    for(int i=0; i<4; i++)
    {
        for(int j=0; j<4; j++)
            cout << board_obj.get_Tetro_Obj().getShapeOfTetris()[i][j];
        cout << endl;
    }
    cout << "And now, i will print type pf tetromino: ";
    cout << board_obj.get_Tetro_Obj().get_type() << endl;
    cout << "And now, i will rotate and print tetromino: \n";
    board_obj.get_Tetro_Obj().rotate('r');
    board_obj.get_Tetro_Obj().print();
    cout << "I will print vertical and horizontal length of board by using getters: \n";
    cout << "Number of row: " << board_obj.GetLenOfRow() << endl;
    cout << "Number of column: " << board_obj.GetLenOfClmn() << endl;
    cout << "I will calculate the vertical length of tetromino\n";
    board_obj.lenght_of_object(len, board_obj.get_Tetro_Obj());
    cout << "Vertical len of tetromino is: " << len << endl;
    cout << "I will add tetromino to the board by using += operator and i will print the board\n";
    cout << "Add function is already implemented inside of += operator.\n";
    board_obj += board_obj.get_Tetro_Obj();
    cout << "I will use move func and it will move 2 unit right. It will also animate. There is sleep because animate will delete terminal\n";
    sleep(3);
    for(int i=0; i<board_obj.GetLenOfRow(); i++)
    {
        for(int j=0; j<board_obj.GetLenOfClmn(); j++)
            board_obj.getBoard()[i][j] = temp.getBoard()[i][j];
    }
    board_obj.move('r', 2, temp);
    return 0;
}